cupy/__init__.py

```diff
@@ -685,13 +685,7 @@ def get_array_module(*args):
 
 disable_experimental_feature_warning = False
 
-
-# set default allocator
-_default_memory_pool = cuda.MemoryPool()
-_default_pinned_memory_pool = cuda.PinnedMemoryPool()
-
-cuda.set_allocator(_default_memory_pool.malloc)
-cuda.set_pinned_memory_allocator(_default_pinned_memory_pool.malloc)
+_default_pinned_memory_pool = None
 
 
 def get_default_memory_pool():
```

Comment:
I think this function does not have to be touched.

Reply:
Should we keep `_default_memory_pool` ?

Reply:
Yes, it doesn't have to with this refactoring, I guess.